From 6ccb2d54bce2eef8e71b93b467c0f49253835c74 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Mon, 14 Aug 2017 16:27:10 +0200
Subject: [PATCH] enforcing fixup

---
 grub-core/genmoddep.awk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/genmoddep.awk b/grub-core/genmoddep.awk
index 8976beb91d0..04c2863e5ab 100644
--- a/grub-core/genmoddep.awk
+++ b/grub-core/genmoddep.awk
@@ -18,7 +18,7 @@ BEGIN {
 
 {
   if ($1 == "defined") {
-    if ($3 in symtab) {
+    if ($3 !~ /^\.refptr\./ && $3 in symtab) {
       printf "%s in %s is duplicated in %s\n", $3, $2, symtab[$3] >"/dev/stderr";
       error++;
     }
