From 5435aaac3c8bfb14bb01a826ec747b783947c86f Mon Sep 17 00:00:00 2001
From: Xuan Guo <nbdd0121>
Date: Mon, 14 Aug 2017 16:27:53 +0200
Subject: [PATCH] Set have_exec to y on cygwin so we have grub_mkrescue.

---
 configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 571f7a0b5aa..c7888e40f66 100644
--- a/configure.ac
+++ b/configure.ac
@@ -204,7 +204,8 @@ case "$host_os" in
 esac
 
 case "$host_os" in
-  cygwin | windows* | mingw32*)	have_exec=n ;;
+  cygwin) have_exec=y ;;
+  windows* | mingw32*) have_exec=n ;;
   aros*) have_exec=n ;;
   *) have_exec=y;;
 esac
