From 6d86c59c24e12c8e8471cdd2affca574fdee160d Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Mon, 26 Jun 2017 12:44:59 -0400
Subject: [PATCH 187/238] don't use int for efi status

---
 grub-core/kern/efi/efi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/kern/efi/efi.c b/grub-core/kern/efi/efi.c
index 866fc20e162..91f75fdb89a 100644
--- a/grub-core/kern/efi/efi.c
+++ b/grub-core/kern/efi/efi.c
@@ -157,7 +157,7 @@ grub_efi_get_loaded_image (grub_efi_handle_t image_handle)
 void
 grub_exit (int retval)
 {
-  int rc = GRUB_EFI_LOAD_ERROR;
+  grub_efi_status_t rc = GRUB_EFI_LOAD_ERROR;
 
   if (retval == 0)
     rc = GRUB_EFI_SUCCESS;
-- 
2.17.0

