From daf2e12adfb61d8c1f7a61322bc2a09bda2335ee Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 11 Jul 2018 13:49:17 -0400
Subject: [PATCH] Make the linuxefi module just be the x86 efi version of linux

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 grub-core/Makefile.core.def | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/grub-core/Makefile.core.def b/grub-core/Makefile.core.def
index 81509b86a0d..524330ee829 100644
--- a/grub-core/Makefile.core.def
+++ b/grub-core/Makefile.core.def
@@ -1710,6 +1710,8 @@ module = {
 module = {
   name = linux;
   x86 = loader/i386/linux.c;
+  x86_efi = loader/i386/efi/linux.c;
+  i386_efi = loader/i386/efi/linux.c;
   xen = loader/i386/xen.c;
   i386_pc = lib/i386/pc/vesa_modes_table.c;
   mips = loader/mips/linux.c;
@@ -1722,8 +1724,11 @@ module = {
   arm64 = loader/efi/linux.c;
   emu = loader/emu/linux.c;
   fdt = lib/fdt.c;
+
   common = loader/linux.c;
   common = lib/cmdline.c;
+
+  efi = loader/efi/linux.c;
 };
 
 module = {
@@ -1784,15 +1789,6 @@ module = {
   enable = x86_64_efi;
 };
 
-module = {
-  name = linuxefi;
-  efi = loader/i386/efi/linux.c;
-  efi = lib/cmdline.c;
-  efi = loader/efi/linux.c;
-  enable = i386_efi;
-  enable = x86_64_efi;
-};
-
 module = {
   name = chain;
   efi = loader/efi/chainloader.c;
