From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 12 Jul 2018 11:00:45 -0400
Subject: [PATCH] Don't build the fdt command.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 grub-core/Makefile.core.def | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/grub-core/Makefile.core.def b/grub-core/Makefile.core.def
index 9039a6f73a3..a6cb7b9009c 100644
--- a/grub-core/Makefile.core.def
+++ b/grub-core/Makefile.core.def
@@ -1729,12 +1729,14 @@ module = {
   efi = loader/efi/linux.c;
 };
 
+/*
 module = {
   name = fdt;
   arm64 = loader/efi/fdt.c;
   common = lib/fdt.c;
   enable = fdt;
 };
+*/
 
 module = {
   name = xnu;
